src/coreclr/jit/lowerxarch.cpp

```diff
@@ -8138,11 +8138,18 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* parentNode, GenTre
         case NI_AVX2_BroadcastScalarToVector256:
         case NI_AVX512F_BroadcastScalarToVector512:
         {
+            var_types baseType = hwintrinsic->GetSimdBaseType();
+            if (baseType == TYP_BYTE || baseType == TYP_UBYTE || baseType == TYP_SHORT || baseType == TYP_USHORT)
```

Comment:
This is better handled as
```suggestion
            if (varTypeIsSmall(baseType))
```